apps/ui/src/components/textfield/__stories__/helpers.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect, useCallback } from 'react';
+import type { ComponentProps, ReactNode } from 'react';
+import { TextField as TextFieldComponent } from '../text-field';
+
+export function useControlledString(initial = '') {
+  const [value, setValue] = useState(initial);
+  return {
+    value,
+    onChange: setValue,
+    setValue,
+  };
+}
+
+export function TextFieldWithState(props: ComponentProps<typeof TextFieldComponent>) {
+  const [localValue, setLocalValue] = useState((props.value as string) ?? '');
+
+  useEffect(() => {
+    if (props.value !== undefined && props.value !== localValue) {
+      setLocalValue(props.value as string);
+    }
+  }, [props.value]);
+
+  const handleChange = useCallback(
+    (v: string) => {
+      setLocalValue(v);
+      props.onChange?.(v as never);
+    },
+    [props],
+  );
+
+  return <TextFieldComponent {...props} value={localValue} onChange={handleChange} />;
+}
+
+export const createIcon = (icon: ReactNode, className?: string) => (
+  <span className={className}>{icon}</span>
+);
+
+export const EMAIL_REGEX = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+export const MAX_BIO_LENGTH = 120;
+
+export interface FormData {
+  firstName: string;
+  lastName: string;
+  email: string;
+  phone: string;
+  bio: string;
+}
+
+export type ValidationErrors = Partial<Record<keyof FormData, string>>;
+
+export function validateForm(form: FormData): ValidationErrors {
```

Comment:
I recommend removing external components. The `validateForm` belongs to external component; so its relationship...